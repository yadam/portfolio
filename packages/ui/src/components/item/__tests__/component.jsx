import React from 'react';
import renderer from 'react-test-renderer';
import Item from '../component';

jest.mock('../../loader', () => 'Loader');

let mockProps;

describe('Item', () => {
  beforeEach(() => {
    mockProps = {
      alt: 'Test alt',
      classes: { card: 'test card className', media: 'test media className' },
      id: 'TestId',
      image: 'TestImageUrl',
      isFunny: undefined,
      loading: false,
      title: 'TestTitle',
    };
  });

  it('renders correctly when loading', () => {
    mockProps.loading = true;
    const tree = renderer.create(<Item {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when finished loading', () => {
    const tree = renderer.create(<Item {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when marked funny', () => {
    mockProps.isFunny = true;
    const tree = renderer.create(<Item {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when marked not funny', () => {
    mockProps.isFunny = false;
    const tree = renderer.create(<Item {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
