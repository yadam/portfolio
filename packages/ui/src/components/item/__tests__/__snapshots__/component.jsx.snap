// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders correctly when finished loading 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 Item-card-1 test card className"
>
  <div
    className="MuiCardContent-root-31"
  >
    <p
      className="MuiTypography-root-32 MuiTypography-subtitle2-51 MuiTypography-paragraph-60"
    >
      TestTitle
    </p>
    <div
      className="MuiCardMedia-root-68 Item-media-2 test media className"
      style={
        Object {
          "backgroundImage": "url(\\"TestImageUrl\\")",
        }
      }
      title="Test alt"
    />
    <h3
      className="MuiTypography-root-32 MuiTypography-h3-46"
    >
      <span
        aria-label="That's Funny!"
        role="img"
      >
        ðŸ˜†
      </span>
    </h3>
    <h3
      className="MuiTypography-root-32 MuiTypography-h3-46"
    >
      <span
        aria-label="Meh..."
        role="img"
      >
        ðŸ˜’
      </span>
    </h3>
  </div>
</div>
`;

exports[`Item renders correctly when loading 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 Item-card-1 test card className"
>
  <div
    className="MuiCardContent-root-31"
  >
    <Loader />
  </div>
</div>
`;

exports[`Item renders correctly when marked funny 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 Item-card-1 test card className"
>
  <div
    className="MuiCardContent-root-31"
  >
    <p
      className="MuiTypography-root-32 MuiTypography-subtitle2-51 MuiTypography-paragraph-60"
    >
      TestTitle
    </p>
    <div
      className="MuiCardMedia-root-68 Item-media-2 test media className"
      style={
        Object {
          "backgroundImage": "url(\\"TestImageUrl\\")",
        }
      }
      title="Test alt"
    />
    <h3
      className="MuiTypography-root-32 MuiTypography-h3-46"
    >
      <span
        aria-label="That's Funny!"
        role="img"
      >
        ðŸ˜†
      </span>
    </h3>
  </div>
</div>
`;

exports[`Item renders correctly when marked not funny 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 Item-card-1 test card className"
>
  <div
    className="MuiCardContent-root-31"
  >
    <p
      className="MuiTypography-root-32 MuiTypography-subtitle2-51 MuiTypography-paragraph-60"
    >
      TestTitle
    </p>
    <div
      className="MuiCardMedia-root-68 Item-media-2 test media className"
      style={
        Object {
          "backgroundImage": "url(\\"TestImageUrl\\")",
        }
      }
      title="Test alt"
    />
    <h3
      className="MuiTypography-root-32 MuiTypography-h3-46"
    >
      <span
        aria-label="Meh..."
        role="img"
      >
        ðŸ˜’
      </span>
    </h3>
  </div>
</div>
`;
